hs/src/Reach/AST/SL.hs

```diff
@@ -177,6 +177,34 @@ data SLVal
   | SLV_ContractCode SrcLoc ConnectorObject
   deriving (Eq, Generic)
 
+isDynamicVal :: SLVal -> Bool
+isDynamicVal = \case
+  SLV_Null {} -> False
+  SLV_Bool {} -> False
+  SLV_Int {} -> False
+  SLV_Bytes {} -> False
+  SLV_BytesDyn {} -> False
+  SLV_String {} -> False
+  SLV_DLC _ -> False
+  SLV_Type {} -> False
+  SLV_Connector {} -> False
+  SLV_Prim {} -> False
+  SLV_Form {} -> False
```

Comment:
These can contain more values, so we have to disallow them

Reply:
Currently, blanket disallowing form and prim will make it so no primitives etc can not be used in nested apps because they are in different universes. I think this can be handled by making `isDynamicPrim` etc. and searching within each prim for dynamic values or changing `sss_universe` to be a `Maybe` and have primitive/form values be `Nothing` to distinguish them. Any preference?

Reply:
I think recurring will have to be the way to do it.